### PR TITLE
Initial version of Docker file with cache

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR introduces a basic Docker file.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+COPY . .
+
+RUN --mount=type=cache,target=$RUSTUP_HOME,from=rust,source=$RUSTUP_HOME \
+    --mount=type=cache,target=$CARGO_HOME,from=rust,source=$CARGO_HOME \
+    --mount=type=cache,target=target \
+    cargo install --root output --path .
+
+
+# Main stage
+FROM debian:bullseye-slim
```
Comment (adlrocha) [inline Dockerfile:26]: Why debian here?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+WORKDIR /app
+
+COPY . .
+
+RUN --mount=type=cache,target=$RUSTUP_HOME,from=rust,source=$RUSTUP_HOME \
+    --mount=type=cache,target=$CARGO_HOME,from=rust,source=$CARGO_HOME \
+    --mount=type=cache,target=target \
+    cargo install --root output --path .
+
```
Comment (adlrocha) [inline Dockerfile:23]: We should also expose whatever port we use for the IPC endpoint. Here you have a sample template, we will have to handle them in tests so we can interact between images: https://github.com/consensus-shipyard/ipc-agent/blob/main/config/template.toml

**Review by adlrocha — APPROVED** [review]
I haven't tried it, if it works let's get this merged and then improve it as we develop the tests.

File: Dockerfile (hunk 1 above)
Comment (dnkolegov) [reply Dockerfile:26]: what should be there? alpine? Debian is a standard choice for Rust

File: Dockerfile (hunk 2 above)
Comment (dnkolegov) [reply Dockerfile:23]: Fixed.

File: Dockerfile (hunk 1 above)
Comment (adlrocha) [reply Dockerfile:26]: Debian is fine for now, we can change it in the future.

**Pull request merged**